cryptoservice/crypto_service.go

```diff
@@ -82,10 +82,15 @@ func (cs *CryptoService) GetPrivateKey(keyID string) (k data.PrivateKey, role st
 	for _, ks := range cs.keyStores {
 		for _, keyPath := range keyPaths {
 			k, role, err = ks.GetKey(keyPath)
-			if err != nil {
+			if err == nil {
+				return
+			}
+			switch err.(type) {
+			case trustmanager.ErrPasswordInvalid, trustmanager.ErrAttemptsExceeded:
```

Comment:
I'm not sure how I feel about this. It's now possible for a key to be in multiple key stores (yubikey, and filestore) with different "passwords" (pin for the yubikey and password for key on disk). Failing to remember the yubikey pin doesn't preclude correctly remembering the password for the file on disk (although I think the passwords are requested at different times, yubikey pin is requested at signing time, file password is requested at `GetPrivateKey` time I think). 

I agree that the error messaging in the current case is pretty poor and mostly this should be solveable by pulling the yubikey, but I don't know if the general case of the private key being stored differently in distinct key stores is going to come up again.

@cyli thoughts?